pkg/action/executor/move_util.go

```diff
@@ -132,6 +250,219 @@ func movePod(client *kclient.Clientset, pod *api.Pod, nodeName string, retryNum
 	return xpod, nil
 }
 
+func deleteInvalidPendingPods(parent *unstructured.Unstructured, podClient v1.PodInterface,
+	podList *api.PodList) {
+	if parent == nil {
+		return
+	}
+	// Fetch the parents pods again
+	newPodList, err := parentsPods(parent, podClient)
+	if err != nil {
+		if err != nil {
+			glog.Errorf("Error getting pod list for %s: %s.", parent.GetKind(), parent.GetName())
+		}
+	}
+
+	// Find the pods which weren't there and if attributed to invalid
+	// turbo-scheduler, delete them.
+	for _, pod := range newPendingInvalidPods(newPodList, podList) {
+		err := commonutil.RetryDuring(defaultRetryLess, defaultRetryShortTimeout,
+			defaultRetrySleepInterval, func() error {
+				return podClient.Delete(pod.Name, &metav1.DeleteOptions{})
+			})
+		if err != nil {
+			glog.Errorf("Failed to delete pending pod: %s.", pod.Name)
+		}
+	}
+
+}
+
+func newPendingInvalidPods(list1, list2 *api.PodList) []*api.Pod {
+	var diffList []*api.Pod
+	for _, pod1 := range list1.Items {
+		found := false
+		for _, pod2 := range list2.Items {
+			if pod1.Name == pod2.Name {
+				found = true
+				break
+			}
+		}
```

Comment:
Nit: Not something critical. Would it be better to convert `list2` into a HashSet of pod names like `existingPodSet` first at the start of this function, and in each iteration of `list1` check if `pod1` name exists in `existingPodSet`? This could improve the time complexity of this func from O(N^2) to O(N) which would be more efficient especially when there are lots of pods we care about here.

Reply:
That might actually be more costly in terms of conversion (from original slice into the set, which is a map) in general. Typically a parent has a few replicas (for example 4) its quite uncommon to see double digit replicas of a workload.